doc/devel/conduct/code_of_conduct.rst

```diff
@@ -0,0 +1,167 @@
+are committed to being a community that everyone enjoys being part of. Although
+we may not always be able to accommodate each individual's preferences, we try
+our best to treat everyone kindly.
+
+No matter how you identify yourself or how others perceive you: we welcome you.
+Though no list can hope to be comprehensive, we explicitly honour diversity in:
+age, culture, ethnicity, genotype, gender identity or expression, language,
+national origin, neurotype, phenotype, political beliefs, profession, race,
+religion, sexual orientation, socioeconomic status, subculture and technical
+ability, to the extent that these do not conflict with this code of conduct.
+
+Though we welcome people fluent in all languages, Matplotlib development is
+conducted in English.
+
+Standards for behaviour in the Matplotlib community are detailed in the Code of
+Conduct above. Participants in our community should uphold these standards
+in all their interactions and help others to do so as well (see next section).
+
+
+Reporting Guidelines
+--------------------
+
+We know that it is painfully common for internet communication to start at or
+devolve into obvious and flagrant abuse.  We also recognize that sometimes
+people may have a bad day, or be unaware of some of the guidelines in this Code
+of Conduct. Please keep this in mind when deciding on how to respond to a
+breach of this Code.
+
+For clearly intentional breaches, report those to the Code of Conduct committee
+(see below). For possibly unintentional breaches, you may reply to the person
+and point out this code of conduct (either in public or in private, whatever is
+most appropriate). If you would prefer not to do that, please feel free to
+report to the Code of Conduct Committee directly, or ask the Committee for
+advice, in confidence.
+
+You can report issues to the Matplotlib Code of Conduct committee, at
+matplotlib@numfocus.org. Currently, the committee consists of:
+
+- Thomas Caswell
+- Hannah Aizenman
+
+If your report involves any members of the committee, or if they feel they have
+a conflict of interest in handling it, then they will recuse themselves from
+considering your report. Alternatively, if for any reason you feel
+uncomfortable making a report to the committee, then you can also contact:
+
+- Senior `NumFOCUS staff <https://numfocus.org/code-of-conduct#persons-responsible>`__: conduct@numfocus.org
+
+
+Incident reporting resolution & Code of Conduct enforcement
+-----------------------------------------------------------
+
+*This section summarizes the most important points, more details can be found
+in* :ref:`CoC_reporting_manual`.
+
+We will investigate and respond to all complaints. The Matplotlib Code of
+Conduct Committee and the Matplotlib Steering Committee (if involved) will
+protect the identity of the reporter, and treat the content of complaints as
+confidential (unless the reporter agrees otherwise).
+
```

Comment:
It might be relevant to keep a track of these reports for repeating behaviours/incidents that would set a precedent when dealing with subsequent reports referring to an individual.

Also, often microaggressions go unnoticed and could lead to significant damage to the community so having a log (report log?) can be useful to identify such behaviours.

Reply:
Reports will be confidential, but I don't think they will disappear upon resolution.  

> Also, often microaggressions go unnoticed and could lead to significant damage to the community so having a log (report log?) can be useful to identify such behaviours.

I'm not quite clear if you mean keep a log of formal CoC complaints, or if you mean keep a further log of "microaggressions" that aren't egregious enough to warrant a full complaint.  The former seems fine, the latter would be pretty problematic.

Reply:
I mean of formal complaints (even if confidential).

This is particularly important for repeating behaviours .
Often certain behaviours / complaints seem rather small when isolated. But they can be quite hurtful and damaging if it's a repeating behaviour or escalating.

Reply:
:+1: That makes sense...